recipe/patches/0001-fix-test-cmake-config.patch

```diff
@@ -0,0 +1,192 @@
+From d76be2b01e920230d06a405529fbc9e79908d8c8 Mon Sep 17 00:00:00 2001
```

Comment:
NB that testing cmake configuration seems pretty central to the functionality of the software and just as a general feeling I think it would be unlikely that a patch of this size would be required to make it work. I could be wrong. It's not needed in [conda-forge's recipe](https://github.com/conda-forge/scikit-build-core-feedstock/tree/main/recipe) and they're on v0.9.10.

Reply:
have you tried removing that CMAKE_GENERATOR env variable and including both make and ninja on all platforms like they have?

Reply:
I'm testing this locally to check!

Reply:
@danpetry https://github.com/scikit-build/scikit-build-core/issues/843#issuecomment-2265741783 

if we unset `CMAKE_GENERATOR` the tests should pass without patch (if we can build, though).

Reply:
(we don't have `make` for `win`)

Reply:
@danpetry @JeanChristopheMorinPerso 
Unsetting `CMAKE_GENERATOR`, is causing test failures, even with the patch. 
Unsetting the patch `0001`  but keeping `CMAKE_GENERATOR` is causing test failures also.

Not exactly sure how to proceed with this, except setting `CMAKE_GENERATOR` and keeping the patch.

Reply:
I didn't really have time to understand the patch in full, but it seems like part of it is removing CMAKE_GENERATOR from the environment so it doesn't affect the tests? so it shouldn't matter whether it's set or not...?

Reply:
@danpetry correct.

The patch is coming from the upstream and it is coming from a merged PR (now): https://github.com/scikit-build/scikit-build-core/pull/847 .

The patch does run the tests with different values of `CMAKE_GENERATOR` mocked (despite the one present in the real environment). Before the patch, the three tests were affected by the external environment variable `CMAKE_GENERATOR`.

Now that the tests pass even if `CMAKE_GENERATOR` is set, we can have it set.

If we set it to Ninja, it works.

As far as I understand though, it is not a good idea having it set. I can guess that is because `scikit-build-core` should be able to pick the correct build tools.

However, without it, our build tools are not found.

See comment https://github.com/AnacondaRecipes/scikit-build-core-feedstock/pull/3#discussion_r1705045323

Reply:
Do you mean that without it, other tests (not the ones addressed in the patch) fail? Because those tests actually do configuration and need the cmake generator to be set, otherwise it uses the default (VS) which isn't found (because we haven't made the vs generator work on our builders)?
i.e. if we don't have the patch, those three tests fail because of a bug concerning the generator env variable changing the cmaker object, and if we don't have the env variable, other tests try to use the vs generator?
(should those other tests be fixed too upstream ideally, not that we really should do this due to time constraints)

Reply:
if that's the case, explaining that in a comment above the env variable would be nice

Reply:
in particular highlighting that the patch addresses different tests from the env variable set in the test section